lib/rust/parser/src/syntax/tree.rs

```diff
@@ -213,7 +209,27 @@ macro_rules! with_ast_definition { ($f:ident ($($args:tt)*)) => { $f! { $($args)
             pub open:  token::Symbol<'s>,
             pub body:  Option<Tree<'s>>,
             pub close: token::Symbol<'s>,
-        }
+        },
+        /// Statement declaring the type of a variable.
+        TypeSignature {
+            /// The variable whose type is being declared.
+            pub variable: token::Ident<'s>,
+            /// The `:` token.
+            pub operator: token::Operator<'s>,
+            /// The variable's type.
+            #[reflect(rename = "type")]
+            pub type_: Tree<'s>,
+        },
+        /// An expression with explicit type information attached.
+        TypeAnnotated {
+            /// The expression whose type is being annotated.
+            pub expression: Tree<'s>,
+            /// The `:` token.
+            pub operator: token::Operator<'s>,
+            /// The expression's type.
+            #[reflect(rename = "type")]
+            pub type_: Tree<'s>,
+        },
```

Comment:
what is the difference between these two?

Reply:
Syntactically? A signature can only be applied to an identifier, and must occur at the top-level of a block; an annotation can be applied to any expression, and can occur anywhere in an expression.